src/software/ai/navigator/path_planner/path_planner_performance_test.cpp

```diff
@@ -0,0 +1,224 @@
+#include <gtest/gtest.h>
+
+#include <iostream>
+#include <typeinfo>
+
+#include "software/ai/navigator/obstacle/obstacle.h"
+#include "software/ai/navigator/obstacle/obstacle_factory.h"
+#include "software/ai/navigator/path_planner/path_planner.h"
+#include "software/ai/navigator/path_planner/straight_line_path_planner.h"
+#include "software/ai/navigator/path_planner/theta_star_path_planner.h"
+#include "software/new_geom/point.h"
+#include "software/new_geom/rectangle.h"
+#include "software/test_util/test_util.h"
+#include "software/world/world.h"
+
+using PathPlannerConstructor = std::function<std::unique_ptr<PathPlanner>()>;
+
+struct PlannerTestCase
+{
+    std::string name = "Unnamed test case";
+    Point start, end;
+    Rectangle navigable_area;
+    std::vector<ObstaclePtr> obstacles;
+    unsigned int num_iterations;
+};
+
+ObstacleFactory obstacle_factory(
+    Util::DynamicParameters->getAIConfig()->getObstacleFactoryConfig());
+
+std::vector<ObstaclePtr> obstacles_10 = {
+    obstacle_factory.createRobotObstacle({0, 0}),
+    obstacle_factory.createRobotObstacle({0, 0.5}),
+    obstacle_factory.createRobotObstacle({0, 1.0}),
+    obstacle_factory.createRobotObstacle({0, 1.5}),
+    obstacle_factory.createRobotObstacle({-0.5, 0}),
+    obstacle_factory.createRobotObstacle({-0.5, 0.5}),
+    obstacle_factory.createRobotObstacle({-0.5, 1.0}),
+    obstacle_factory.createRobotObstacle({-0.5, 1.5}),
+    obstacle_factory.createRobotObstacle({0.5, 1.0}),
+    obstacle_factory.createRobotObstacle({0.5, 1.5}),
+};
+
+std::vector<ObstaclePtr> obstacles_20 = {
+    obstacle_factory.createRobotObstacle({0, 0}),
```

Comment:
non blocking nit:

Since it seems like each obstacle vector (at least mostly) builds off the previous ones, maybe it would be easier to write
```
obstacles_20 = obstacles_10.append({...})
```

Reply:
cleaned that up